xroad-catalog-collector/src/main/java/fi/vrk/xroad/catalog/collector/util/OrganizationUtil.java

```diff
@@ -50,13 +50,14 @@ private OrganizationUtil() {
 
     }
 
-    public static List<String> getOrganizationIdsList(String url)
+    public static List<String> getOrganizationIdsList(String url, Integer fetchOrganizationsLimit)
             throws KeyStoreException, NoSuchAlgorithmException, KeyManagementException {
         String response = getResponseBody(url);
         JSONObject json = new JSONObject(response);
         JSONArray itemList = json.optJSONArray("itemList");
         List<String> idsList = new ArrayList<>();
-        for (int i = 0; i < itemList.length(); i++) {
+        int totalFetchAmount = itemList.length() > fetchOrganizationsLimit ? fetchOrganizationsLimit : itemList.length();
```

Comment:
I don't understand the advantage of fetchOrganizationsLimit. Doesn't the itemList already have all the items even if there were one billion of them? Should/could the limit be applied before actual request to the url?

Reply:
The itemList consist of 716 items, the parameter currently is set to 2000, in order not to limit the fetch amout, but the parameter value can be changed to not fetch all the organizations. I use this parameter in FetchOrganizationsActorTest for example, otherwise this test would take 10 minutes to run.

Reply:
The ItemList is request with the first request from the API and this can not be limited as the request is about fetching just all the IDs of all the organizations, this request does not take long, up to 30 seconds. Now, another request needs to be done with these IDs, in order to request the organization data. Maximum amount of ids to use per request is 100. This request we can limit with the parameter, so we could only use for example 100 of these 716 organization ids.